providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/e2e/steps/config/ConfigSteps.java

```diff
@@ -0,0 +1,103 @@
+package dev.openfeature.contrib.providers.flagd.e2e.steps.config;
+
+import dev.openfeature.contrib.providers.flagd.Config;
+import dev.openfeature.contrib.providers.flagd.FlagdOptions;
+import dev.openfeature.contrib.providers.flagd.resolver.grpc.cache.CacheType;
+import io.cucumber.java.en.Then;
+import io.cucumber.java.en.When;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class ConfigSteps {
+
+    FlagdOptions.FlagdOptionsBuilder builder = FlagdOptions.builder();
+    FlagdOptions options;
+
+    @When("we initialize a config")
+    public void we_initialize_a_config() {
+        options = builder.build();
+    }
+
+    @When("we initialize a config for {string}")
+    public void we_initialize_a_config_for(String string) {
+        switch (string.toLowerCase()) {
+            case "in-process":
+                options = builder.resolverType(Config.Resolver.IN_PROCESS).build();
+                break;
+            case "rpc":
+                options = builder.resolverType(Config.Resolver.RPC).build();
+                break;
+            default:
+                throw new RuntimeException("Unknown resolver type: " + string);
+        }
+    }
+
+    @When("we have an option {string} of type {string} with value {string}")
+    public void we_have_an_option_of_type_with_value(String option, String type, String value) throws ClassNotFoundException, NoSuchMethodException, InvocationTargetException, IllegalAccessException {
+        Object converted = convert(value, type);
+        Method method = Arrays.stream(builder.getClass().getMethods())
+                .filter(method1 -> method1.getName().equals(option))
+                .findFirst()
+                .orElseThrow(RuntimeException::new);
+        method.invoke(builder, converted);
+    }
+
+
+    Map<String, String> envVarsSet = new HashMap<>();
+
+    @When("we have an environment variable {string} with value {string}")
+    public void we_have_an_environment_variable_with_value(String string, String string2) throws IllegalAccessException, NoSuchFieldException {
```

Comment:
```suggestion
    public void we_have_an_environment_variable_with_value(String string, String string2) throws IllegalAccessException, NoSuchFieldException {
```

I'd prefer if we update the param names here to be more descriptive instead of the auto-generated `string1/2`.